packages/roosterjs-editor-api/lib/format/clearFormat.ts

```diff
@@ -46,10 +46,34 @@ export default function clearFormat(editor: Editor) {
                 setFontSize(editor, defaultFormat.fontSize);
             }
             if (defaultFormat.textColor) {
-                setTextColor(editor, defaultFormat.textColor);
+                if (defaultFormat.originalSourceTextColor) {
+                    const colorPair = {
+                        lightModeColor: editor.isDarkMode() ?
+                        defaultFormat.originalSourceTextColor :
+                        defaultFormat.textColor,
+                        darkModeColor: !editor.isDarkMode() ?
+                        defaultFormat.originalSourceTextColor :
+                        defaultFormat.textColor
+                    }

```

Comment:
The value of defaultFormat's colors based on if isDarkMode is set or not seems risky to me. It's the same scenario where callers need to be aware of the current state of the editor (see here).

In this setup, `originalSource<Foo>` also seems to actually refer to `otherMode<Foo>`.

Since we're worried about backwards compatibility, would something like this work:

```typescript
defaultFormat = {
  textColors: {
    lightModeColor: ...
    darkModeColor: ...
  }
  get textColor() {
    return editor.isDarkMode()
      ? this.textColors.darkModeColor
      : this.textColors.lightModeColor
  }
  set textColor(value: string) {
    this.textColors = editor.isDarkMode()
      ? colorPairFromDarkmodeColor(value)
      : colorPairFromLightmodeColor(value)
  }
}
```

Reply:
I may change this entirely, as the more I look at it the more dis-satisfied I am with the result.

My original thinking was that DarkMode would use a seperate default format, and, in the future, if we enabled switching between the two modes, we would swap between one and the other.

OriginalSource as a concept has also been clobbered to mean "other mode" as Max highlighted.

A DefaultFormat should provide all color mappings applicable. If, for whatever reason, they aren't defined, we should provide our defaults, just like before. No more separate dark mode default format to prevent over complication.

I'm gonna take another pass at this, taking into consideration the proposed changes above, and try to arrive at something that makes sense.